### PR TITLE
Interpolate ClientHeaderParam in REST client reactive

### LINKED ISSUE #30631: allow additional text in clientheader
### Description

clientheaderparam cannot include additional text outside the ${}.

this fails:

```
@POST
    @ClientHeaderParam(name = "Authorization", value = "Bearer ${openai.token}")
    Set<String> getCompletions(Completion payload);
```

where as this works but now I have to include "Bearer " in the property:

```
@POST
    @ClientHeaderParam(name = "Authorization", value = "${openai.token}")
    Set<String> getCompletions(Completion payload);
```

if possible to do without  overhead this seems like a nice touch/simplification. 

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by gastaldi**
- Add interpolation functionality for expressions of the form `${config.property.name}`
- Add a test for supporting additional text on the header property
- Fixes #30631

### CODE REVIEW

File: extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/headers/ClientHeaderParamFromPropertyTest.java [hunk 1]
```diff
@@ -78,5 +93,11 @@ public interface Client {
         @GET
         @ClientHeaderParam(name = "some-other-header", value = "${non-existent-property}", required = false)
         String missingNonRequiredProperty();
+
+        @GET
+        @ClientHeaderParam(name = "additional-text", value = "Bearer ${my.property-value}")
```
Comment (geoand) [inline extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/headers/ClientHeaderParamFromPropertyTest.java:98]: How about we add some text after `}` as well?

Comment (gastaldi) [reply extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/headers/ClientHeaderParamFromPropertyTest.java]: Good idea, done

**Review by geoand — APPROVED** [review]

File: extensions/resteasy-reactive/rest-client-reactive/runtime/src/main/java/io/quarkus/rest/client/reactive/runtime/ConfigUtils.java [hunk 2]
```diff
@@ -9,20 +9,51 @@ public class ConfigUtils {
...
+        while ((idx = sb.lastIndexOf(PREFIX)) > -1) {
+            int endIdx = sb.indexOf(SUFFIX, idx);
+            if (endIdx < 0) {
+                throw new IllegalArgumentException("Invalid expression: " + expression);
+            }
+            String configValue = getConfigValue(sb.substring(idx, endIdx + 1), required);
+            // If no value is found, we return null directly
+            if (configValue == null) {
+                return null;
+            }
```
Comment (gastaldi) [inline extensions/resteasy-reactive/rest-client-reactive/runtime/src/main/java/io/quarkus/rest/client/reactive/runtime/ConfigUtils.java:34]: @geoand I added this after the last review. The idea is that if the config property is not found, the expression itself is null (not used). 

Eg. `Bearer ${property}` - if `${property}` is not set then the header parameter is not set

Comment (geoand) [reply extensions/resteasy-reactive/rest-client-reactive/runtime/src/main/java/io/quarkus/rest/client/reactive/runtime/ConfigUtils.java:34]: 👍

**Pull request merged**